DRYD-1818: Implement image ordering functionality for public browser

spirosdi opened the pull request:
**What does this do?**
* Changed query for getting medias from searching objects to searching medias filtered by object id and sorted by sort param.
* The sorting is configurable.
* Changed storing title in state when quering for medias. Set using the already existing title in state for images alt text.

**Why are we doing this? (with JIRA link)**
We need to be able to search for medias, filtered by related objects and sorted by updatedAt, identificationNumber or title. https://collectionspace.atlassian.net/browse/DRYD-1818

**How should this be tested? Do these changes have associated tests?**
First you need to deploy services application after checking out https://github.com/collectionspace/services/pull/479.
Then in public browser:
1. Open a detail view of an object
2. Check that the images in the gallery are sorted according to the sorting configuration
3. Check that for images without alt text set, the object title (if existing) is being used for alt text

**Dependencies for merging? Releasing to production?**
The 
https://github.com/collectionspace/services/pull/479 
https://github.com/collectionspace/cspace-public-gateway/pull/33
should be first merged and released.

**Has the [public browser documentation](https://collectionspace.atlassian.net/wiki/spaces/COL/pages/666274513/Public+Collections+Browser) been updated for these changes?**
No, only the dev documentation has been updated. We should decide if it makes sense to create links from wiki documentation to the github ones.

**Did someone actually run this code to verify it works?**
@spirosdi ran it locally

spirosdi committed:
DRYD-1818: replaced fetching medias by searching objects with searching medias filtered by object and sorted by sort param; replaced getting title used in alt text from API response with getting it from data existing in state;

spirosdi committed:
DRYD-1818: updated documentation; changed format of mediaSnapshotSort config; changed default value;

spirosdi committed:
DRYD-1818: fixed getting sortDirection from config; fixed getting blobcsid;

mikejritter commented on src/actions/mediaActions.js:
```diff
@@ -35,17 +41,21 @@ export const findMedia = (referenceValue, institutionId) => (dispatch, getState)
 
   const query = {
-    _source: ['collectionspace_denorm:mediaCsid', 'collectionspace_denorm:mediaAltText', 'collectionspace_denorm:title'],
+    _source: ['media_common:blobCsid', 'media_common:altText'],
```
I noticed I wasn't seeing any images when testing this locally, and I believe this should use `ecm:name` instead of `blobCsid` in order to get the derivatives back. The API call (which the gateway proxies) looks like `/cspace-services/media/${mediaCsid}/blob/derivatives/${size}/content`.

spirosdi replied:
True, I fixed it to use the `referenceField` from config. I was using an improvised setup for running public browser locally, that was bypassing the gateway completely. But now I set it correct and am able to reproduce the issue. Thanks for catching!

mikejritter commented on src/actions/mediaActions.js:
```diff
@@ -55,11 +65,10 @@ export const findMedia = (referenceValue, institutionId) => (dispatch, getState)
-      const mediaCsids = get(source, 'collectionspace_denorm:mediaCsid') || [];
-      const mediaAltTexts = get(source, 'collectionspace_denorm:mediaAltText') || [];
+      const hits = get(data, ['hits', 'hits'], []);
+      const mediaCsids = hits.map((hit) => get(hit, ['_source', 'media_common:blobCsid'], ''));
```
Same as above, this would also need to be changed to `ecm:name`

mikejritter reviewed (changes requested):
@spirosdi just one issue that I noticed while testing. We'll need to update the gateway documentation as well to note that the `allowedRecordTypes` will need to include `Media` now.

spirosdi committed:
DRYD-1818: replaced using blobCsid with ecm:name for storing in mediaCsids

spirosdi commented:
> @spirosdi just one issue that I noticed while testing. We'll need to update the gateway documentation as well to note that the `allowedRecordTypes` will need to include `Media` now.

That's right. I missed that due to my improvised local setup for running public-browser locally. It was bypassing gateway completely. 

Anyway, in the end we should add `Media` in default `allowedRecordTypes`. I added a gateway PR here: https://github.com/collectionspace/cspace-public-gateway/pull/32. I added the PR as a dependency of this one.

mikejritter reviewed (approved):
Everything is working now :+1:
